### PR TITLE
Add script to produce source code zip of browser extension

### LINKED ISSUE #12918: Produce a source code zip of Firefox add-on to faciliate compliance review
The Mozilla add-on team needs access to the source code of our Firefox add-on in order to be able review it for compliance with their policy. We need to submit this source code with every new version of the add-on.

This issue is to track the requirements for this source code.

The source code zip:

- **must** contain the exact snapshot that was used to produce the submitted add-on (XPI file)
- **must** contain all scripts required to reproduce the build (although it can pull/build dependencies with `yarn install` etc)
- **should** reproduce the build exactly, so that a diff of the result shows no difference (before zipping; because after zipping the XPI will actually be non-identical due to file timestamps being included inside the zip metadata).
- **should** only include source code that pertains to the add-on, and not the entirety of `sourcegraph/sourcegraph`. This means the solution is not to simply zip the entire repository.
- **should**, for our convenience, be something that can be done automatically by a script
- **could** if possible, in addition to producing the source code zip, upload it automatically with the submitted XPI.

### PR DESCRIPTION
**Pull request by marekweb**
**Background**

For publishing our extensions, we need to provide source code for review. The Firefox add-on registry requires this before publishing. An outline of these requirements is described in #12918 

Until now the process of producing a source code zip was manual.

**Implementation**

This introduces a script: `yarn run create-source-zip` which produces `sourcegraph.zip` as an output.

The script's configuration specifies the particular commit of `sourcegraph/sourcegraph` to use, which is fetched from the repository.

By default the script also fetches and includes `sourcegraph/code-intel-extensions` for bundling in the Firefox build.

**Related issues**

Closes #12918 because this PR implements most of the scope of that issue.

The only remaining scope is automating this process in CI, which has been previously filed under #11370.

### CODE REVIEW

**Review by tjkandala — APPROVED** [review]

File: browser/scripts/create-source-zip.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * used to produce a build of the browser extensions.
+ *
+ * This makes it possible to provide the complete source code to the maintainers of
+ * extension registries, in the cases where providing the source code is a
+ * pre-requisite to getting approval for publishing.
+ *
+ * This script fetches a fresh copy of the repository, identified by commit ID.
+ * This is done instead of using packaging the contents of the current working copy
+ * repository, and the reason is to avoid including any build artifacts that may
+ * be present. By pinning it to a commit, it become
```
Comment (tjkandala) [inline browser/scripts/create-source-zip.js:15]: unfinished doc comment?

Comment (marekweb) [reply browser/scripts/create-source-zip.js]: Thanks for catching that @tjkandala

File: browser/scripts/create-source-zip.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * This script fetches a fresh copy of the repository, identified by commit ID.
+ * This is done instead of using packaging the contents of the current working
+ * copy repository, and the reason is to avoid including any build artifacts
+ * that may be present. By pinning it to a commit, it's possible to use this
+ * script across any branch and commit.
+ *
+ */
+
+// Configuration
+const commitId = '350764282631014ea24ccd88fda459a4b19a5669'
```
Comment (felixfbecker) [inline browser/scripts/create-source-zip.js:21]: **suggestion**: we could get this from `git rev-parse HEAD`

**Review by felixfbecker — COMMENTED** [review]
Awesome!

**Pull request merged**